math/src/main/java/com/ivy/math/calculator/AppendDecimalSeparator.kt

```diff
@@ -3,8 +3,12 @@ package com.ivy.math.calculator
 fun appendDecimalSeparator(
     expression: String, decimalSeparator: Char
 ): String {
+    fun alreadyAddedDecimal(expression: String): Boolean =
+        expression.split(Regex("[^0-9^${decimalSeparator}]")).last()
```

Comment:
A short comment explaining the logic behind the regex would be a great addition!
For example:
```
// 1.01 + 1.01 split by ... because ...
```